Mathlib/Tactic/CancelDenoms/Core.lean

```diff
@@ -259,15 +259,18 @@ def derive (e : Expr) : MetaM (ℕ × Expr) := do
 In the case of `LT`, `LE`, `GE`, and `GT` an order on the type is needed, in the last case
 it is not, the final component of the return value tracks this.
 -/
-def findCompLemma (e : Expr) : Option (Expr × Expr × Name × Bool) :=
-  match e.getAppFnArgs with
-  | (``LT.lt, #[_, _, a, b]) => (a, b, ``cancel_factors_lt, true)
-  | (``LE.le, #[_, _, a, b]) => (a, b, ``cancel_factors_le, true)
-  | (``Eq, #[_, a, b]) => (a, b, ``cancel_factors_eq, false)
-  | (``Ne, #[_, a, b]) => (a, b, ``cancel_factors_ne, false)
-  | (``GE.ge, #[_, _, a, b]) => (b, a, ``cancel_factors_le, true)
-  | (``GT.gt, #[_, _, a, b]) => (b, a, ``cancel_factors_lt, true)
-  | _ => none
+def findCompLemma (e : Expr) : MetaM (Option (Expr × Expr × Name × Bool)) := do
+  match (← whnfR e).getAppFnArgs with
+  | (``LT.lt, #[_, _, a, b]) => return (a, b, ``cancel_factors_lt, true)
+  | (``LE.le, #[_, _, a, b]) => return (a, b, ``cancel_factors_le, true)
+  | (``Eq, #[_, a, b]) => return (a, b, ``cancel_factors_eq, false)
+  -- `a ≠ b` reduces to `¬ a = b` under `whnf`
```

Comment:
Can you add a test for this? This ought to do:
```
example : ¬ (4 / 2 : ℚ) = 3 := by cancel_denoms; norm_num
```
which previously failed.

Reply:
Added!